CMakeLists.txt

```diff
@@ -1,21 +1,26 @@
-cmake_minimum_required(VERSION 3.3)
+cmake_minimum_required(VERSION 3.12)
 #set(CMAKE_C_COMPILER "gcc-7")
 #set(CMAKE_CXX_COMPILER "g++-7")
 project(cosoco)
 
+# Check if the 'no-xcsp3' option is enabled
+option(NO_XCSP3 "Disable XCSP3 support" OFF)
 
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/main")
```

Comment:
The definition of those path are now set to the build.sh file. That way, we can safely build the different versions in different folders without having to fear that one might contaminate the other.